### PR TITLE
Set vim options instead of sniffing the config

### LINKED ISSUE #2317: Suggested method for "Securing Your Editor" still leaks passwords.
The suggested method here https://github.com/gopasspw/gopass/blob/master/docs/setup.md#securing-your-editor, which is adding
```
au BufNewFile,BufRead /dev/shm/gopass.* setlocal noswapfile nobackup noundofile
```

to the vimrc, is not enough. The viminfo-file still gets written with content of the decrypted file. What should be used instead is
```
autocmd BufNewFile,BufRead /dev/shm/gopass* setlocal noswapfile nobackup noundofile viminfo=""
```
for vim and
```
autocmd BufNewFile,BufRead /dev/shm/gopass* setlocal noswapfile nobackup noundofile shada=""
```
for neovim.
So the path is wrong (the dot at the end) and it is still writing viminfo/shada.
I do not know if the MacOS path is correct, but the viminfo/shada-stuff is of course similar.

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #2317

RELEASE_NOTES=[ENHANCEMENT] Set vim options instead of sniffing

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: internal/editor/editor.go [hunk 1]
```diff
@@ -128,3 +86,25 @@ func Invoke(ctx context.Context, editor string, content []byte) ([]byte, error)
...
+	viminfo := `viminfo=""`
+	if editor == "neovim" {
+		viminfo = `shada=""`
+	}
+
+	return []string{
+		"-i", "NONE", // disable viminfo
+		"-n", // disable swap
+		"-c",
+		fmt.Sprintf("autocmd BufNewFile,BufRead %s setlocal noswapfile nobackup noundofile %s", path, viminfo),
```
Comment (AnomalRoil) [inline internal/editor/editor.go:108]: I guess we might want to document the fact that we're setting these somewhere in the documentation, no?

File: internal/editor/editor.go [hunk 2]
```diff
@@ -128,3 +86,25 @@ func Invoke(ctx context.Context, editor string, content []byte) ([]byte, error)
...
 	return nContent, nil
 }
+
+func vimOptions(editor string) []string {
+	if editor != "vi" && editor != "vim" && editor != "neovim" {
+		return []string{}
+	}
+
+	path := "/dev/shm/gopass*"
+	if runtime.GOOS == "darwin" {
```
Comment (AnomalRoil) [inline internal/editor/editor.go:96]: How does that work on Windows? (Are we even using vim on windows?)

File: internal/editor/editor.go [hunk 3]
```diff
@@ -27,47 +22,9 @@ var (
...
-
-	if vimOptsRe.Match(buf) {
-		debug.Log("Recommended settings found in %s", vrc)
-
-		return nil
-	}
-
-	debug.Log("%s did not match %s", string(buf), vimOptsRe)
-	out.Warningf(ctx, "Vim might leak credentials. Check your setup.\nhttps://go.gopass.pw/setup#securing-your-editor")
-
```
Comment (AnomalRoil) [inline internal/editor/editor.go:67]: Shouldn't we also change this documentation entry to reflect the new behaviour?

**Review by AnomalRoil — DISMISSED** [review]

File: internal/editor/editor.go (hunk 2 above)
Comment (dominikschulz) [reply internal/editor/editor.go:96]: I have no idea. Someone else would need to test that.

**Pull request merged**